cpp/oneapi/dal/test/ccl_communicator.cpp

```diff
@@ -241,6 +241,62 @@ TEST_M(ccl_comm_test, "allgatherv") {
     }
 }
 
+TEST_M(ccl_comm_test, "allgatherv_arbitrary_displacements") {
+    auto comm = get_new_comm();
+    const std::int64_t granularity = 10;
+    const std::int64_t rank_count = comm.get_rank_count();
+    const std::int64_t rank = comm.get_rank();
+
+    std::vector<std::int64_t> recv_counts(rank_count);
+    std::vector<std::int64_t> displs(rank_count);
+    std::int64_t total_size = 0;
+    for (std::int64_t i = 0; i < rank_count; i++) {
+        recv_counts[i] = (i + 2) * granularity;
```

Comment:
The idea is not to increase receive count for every rank but to add a gap between results obtained from different ranks, Please also add a constant for gap width. Initial array should be filled by -1 (or any other values that are different against collected values) and values inside gaps should remain unchanged after allgatherv call.